TYPO3.Flow/Classes/TYPO3/Flow/Utility/Environment.php

```diff
@@ -58,28 +47,22 @@ public function __construct(\TYPO3\Flow\Core\ApplicationContext $context)
      *
      * @param string $temporaryDirectoryBase Base path of the temporary directory, with trailing slash
      * @return void
+     * @throws Exception
+     * @deprecated since 2.3 - Set the environment variable FLOW_PATH_TEMPORARY_BASE to change the temporary directory base, see Bootstrap::defineConstants()
      */
     public function setTemporaryDirectoryBase($temporaryDirectoryBase)
     {
-        $this->temporaryDirectoryBase = $temporaryDirectoryBase;
-        $this->temporaryDirectory = null;
+        throw new Exception('Changing the temporary directory path during runtime is no longer supported. Set the environment variable FLOW_PATH_TEMPORARY_BASE to change the temporary directory base', 1441355116);
     }
 
     /**
      * Returns the full path to Flow's temporary directory.
      *
-     * @return string Path to PHP's temporary directory
-     * @api
+     * @return string
      */
     public function getPathToTemporaryDirectory()
     {
-        if ($this->temporaryDirectory !== null) {
-            return $this->temporaryDirectory;
-        }
-
-        $this->temporaryDirectory = $this->createTemporaryDirectory($this->temporaryDirectoryBase);
```

Comment:
Whoa, I don't like this… Before, when the temporary directory could not be created a helpful message was thrown, not Flow will probably just die in weird ways. What about checking the needed checks in `defineConstants()` at least?